### PR TITLE
Add Azure Bastion as the jumpbox recommendation

### PR DESCRIPTION
**Pull request by MikeBazMSFT**
The traditional VM-based jumpbox is no longer recommended; Azure Bastion is recommended.  Some of the other architectures eg https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/dmz/secure-vnet-dmz?tabs=portal are already there.

This PR keeps the information on how to secure a traditional VM jumpbox but adds Azure Bastion as well.  A reasonable alternative would be to remove all references to a traditional jumpbox VM but I didn't know if we wanted to lose that security information here so I kept it for now.

### CODE REVIEW

File: docs/reference-architectures/n-tier/n-tier-sql-server-content.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ The architecture has the following components.
...
 
-- **Jumpbox**. Also called a [bastion host]. A secure VM on the network that administrators use to connect to the other VMs. The jumpbox has an NSG that allows remote traffic only from public IP addresses on a safe list. The NSG should permit remote desktop (RDP) traffic.
+- **Jumpbox**. Also called a [bastion host]. Traditionally, a secure VM on the network that administrators use to connect to the other VMs. The jumpbox has an NSG that allows remote traffic only from public IP addresses on a safe list. The NSG should permit Remote Desktop Protocol (RDP) traffic. Azure offers the managed solution [Azure Bastion](/azure/bastion/) to meet this need.
```
Comment (tiburd) [inline docs/reference-architectures/n-tier/n-tier-sql-server-content.md:42]:
```suggestion
- **Jumpbox**. Also called a [bastion host]. Traditionally, a secure VM on the network that administrators use to connect to the other VMs. The jumpbox has an NSG that allows remote traffic only from public IP addresses on a safe list. The NSG should permit Remote Desktop Protocol (RDP) traffic. Azure offers the managed solution [Azure Bastion](/azure/bastion/bastion-overview) to meet this need.
```

File: docs/reference-architectures/n-tier/n-tier-sql-server-content.md [hunk 2]
```diff
@@ -107,11 +107,15 @@ Test your deployment by [forcing a manual failover][sql-alwayson-force-failover]
...
+- Don't allow RDP access from the public internet to the VMs that run the application workload. Instead, all RDP access to these VMs should go through the jumpbox. An administrator logs in to the jumpbox and then logs in to the other VM from the jumpbox. The jumpbox allows RDP traffic from the internet, but only from known, safe IP addresses.
+- The jumpbox has minimal performance requirements, so select a small VM size. Create a [public IP address] for the jumpbox. Place the jumpbox in the same virtual network as the other VMs, but in a separate management subnet.
+- To secure the jumpbox, add an NSG rule that allows RDP connections only from a safe set of public IP addresses. Configure the NSGs for the other subnets to allow RDP traffic from the management subnet.
+
+For a customer-managed VM, all these rules apply. However, the current recommendation is to use [Azure Bastion](/azure/bastion/), a managed jumpbox solution that allows for HTML5 access to RDP or SSH behind Azure AD protection. This is a much simpler solution that ultimately has a lower total cost of ownership (TCO) for the customer.
```
Comment (tiburd) [inline docs/reference-architectures/n-tier/n-tier-sql-server-content.md:118]:
```suggestion
For a customer-managed VM, all these rules apply. However, the current recommendation is to use [Azure Bastion](/azure/bastion/bastion-overview), a managed jumpbox solution that allows for HTML5 access to RDP or SSH behind Azure AD protection. This is a much simpler solution that ultimately has a lower total cost of ownership (TCO) for the customer.
```

File: docs/reference-architectures/n-tier/n-tier-sql-server-content.md [hunk 3]
```diff
@@ -107,11 +107,15 @@ Test your deployment by [forcing a manual failover][sql-alwayson-force-failover]
...
 
-The jumpbox has minimal performance requirements, so select a small VM size. Create a [public IP address] for the jumpbox. Place the jumpbox in the same virtual network as the other VMs, but in a separate management subnet.
+In the past, a VM that's managed by the customer might be used as a jumpbox. In that scenario, the following recommendations apply:
 
-To secure the jumpbox, add an NSG rule that allows RDP connections only from a safe set of public IP addresses. Configure the NSGs for the other subnets to allow RDP traffic from the management subnet.
+- Don't allow RDP access from the public internet to the VMs that run the application workload. Instead, all RDP access to these VMs should go through the jumpbox. An administrator logs in to the jumpbox and then logs in to the other VM from the jumpbox. The jumpbox allows RDP traffic from the internet, but only from known, safe IP addresses.
+- The jumpbox has minimal performance requirements, so select a small VM size. Create a [public IP address] for the jumpbox. Place the jumpbox in the same virtual network as the other VMs, but in a separate management subnet.
```
Comment (tiburd) [inline docs/reference-architectures/n-tier/n-tier-sql-server-content.md:115]: Why is "public IP address" in brackets?

Comment (MikeBazMSFT) [reply docs/reference-architectures/n-tier/n-tier-sql-server-content.md:115]: I didn't change that, it was there before.  In the live article it links to the public IP addresses article and I didn't see a need to change it.  This whole section really was just changed into a list to flow better with the addition of Azure Bastion but that's the only change to that text I made.

**Review by EdPrice-MSFT — APPROVED** [review]
Great updates

**Comment by EdPrice-MSFT** [comment]
#Sign-Off

**Comment by PRMerger12** [comment]
@EdPrice-MSFT: I'm sorry - only the author of this article, @, can sign off on your changes. But we do have an exception process - if you are on the Microsoft content or product team for this product area, you can ask the PR review team to review and merge it by sending mail to the techdocprs alias.

**Comment by EdPrice-MSFT** [comment]
@ktoliver - We finished our review. How can we sign off?

**Comment by ktoliver** [comment]
> We finished our review. How can we sign off?

I'll find out whether there's something Engineering can do to resolve the issue of automation failing because metadata isn't in the file that's updated in the PR.

**Comment by MikeBazMSFT** [comment]
officially this is merged but it isn't published on the live site; is that because of the automation failure you're referring to, @ktoliver ?

**Comment by ktoliver** [comment]
> officially this is merged but it isn't published on the live site; is that because of the automation failure you're referring to, @ktoliver ?

@MikeBazMSFT No, the automation I referred to is when PRMerger looks for an author to assign a newly created pull request to for technical review, and it generates an accompanying notification email. 

I'll look into why the update isn't showing on the live site yet. I'll be back with an answer.

**Comment by ktoliver** [comment]
@MicrosoftDocs/pubdesk - Would you be able to take a look at why updates merged in this PR aren't live? They *are* in master in the private repo, and there have been no subsequent commits that would have overwritten what was merged here. I can't see what prevented the updates from merging to the live branch. 

Updates were merged in PR in the public repo:
https://github.com/MicrosoftDocs/architecture-center/pull/2900/files

Updates are in master in the private repo, so the repo sync was successful:
https://github.com/MicrosoftDocs/architecture-center-pr/blob/master/docs/reference-architectures/n-tier/n-tier-sql-server-content.md (internal link)

However, the updates aren’t seen on the live site:
https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/n-tier/n-tier-sql-server

Thanks. Do you need a ticket opened to look into this? 

cc @tiburd

**Comment by ktoliver** [comment]
@MikeBazMSFT I'm seeing the revisions in this PR live now: https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/n-tier/n-tier-sql-server

**Comment by MikeBazMSFT** [comment]
As am I! 🥳

**Pull request merged**